client/dynamicplugins/registry.go

```diff
@@ -31,16 +31,48 @@ type Registry interface {
 	StubDispenserForType(ptype string, dispenser PluginDispenser)
 }
 
+// RegistryState is what we persist in the client state store. It contains
+// a map of plugin types to maps of plugin name -> PluginInfo.
+type RegistryState struct {
```

Comment:
I tried to do this as a type alias like `type RegistryState map[string]map[string]*PluginInfo` but initializing it created a ton of extra work in all the consumers, whereas we can just do one `make` in `NewRegistry` here and be done with it.